packages/vite/src/node/plugins/oxc.ts

```diff
@@ -176,8 +195,19 @@ export function oxcPlugin(config: ResolvedConfig): Plugin {
       // recycle serve to avoid preventing Node self-exit (#6815)
       setServer(null)
     },
-    async transform(code, id) {
+    async transform(code, id, options) {
       if (filter(id) || filter(cleanUrl(id))) {
+        // disable refresh at ssr
+        if (options?.ssr && oxcTransformOptions.jsx?.refresh) {
+          oxcTransformOptions.jsx.refresh = false
+        }
```

Comment:
Just to confirm, this `refresh` option is only for React and won't be used by other JSX frameworks, right? (If it can be used then, I guess we cannot use the assumption that refresh should be `false` for ssr.)

Reply:
I'm not sure other JSX frameworks hmr how to works. Let me check it.

Reply:
- preact using `Prefresh-babel`
- solid using `solid-refresh/babel`

I think it is only for react at now. We could change it if here found other cases.